### ISSUE TITLE
When a feature has 2 scenarios that share steps, it always runs the code of the first scenario.

### ISSUE THREAD

**Issue by damasofc** [TheBrainFamily/cypress-cucumber-preprocessor#435]
### Current behavior
I've this feature:

~~~
Feature: Register
    Scenario: Register New User - Invalid Fields
        Given A user wants to register
        When filling in the required information
        But some fields contain invalid information
        Then an error message will appear
        And the user will remain in the 'current' page
    Scenario: Register New User - Valid Fields
        Given A user wants to register
        When filling in the required information
        And all fields contain valid information
        Then load the "confirmation" page
~~~

And I have 2 files for each scenario in the folder /integration/Register

The first file for the Invalid Fields is this one:

~~~
import { Given,When,Then,But,And, Before } from "cypress-cucumber-preprocessor/steps";
import LoginPage from '../../pages/LoginPage'
import RegisterPage from '../../pages/RegisterPage'

Given('A user wants to register', () => {
    RegisterPage.visit();
});

When('filling in the required information', () => {
    RegisterPage.fillData('testing@hotmail.com','do123','do123');
});

But('some fields contain invalid information', () => {
    RegisterPage.fillData('d','d123','d123');
});

Then('an error message will appear', () => {
    RegisterPage.fillData('test@hotmail.com','d','d');
});

And("the user will remain in the 'current' page", () => {
    RegisterPage.fillData('dac@.com','dam','damaso123');
})
~~~

and this is the other file for the valid fields

~~~
import { Given,When,Then,But,And, Before } from "cypress-cucumber-preprocessor/steps";
import RegisterPage from '../../pages/RegisterPage'

Given('A user wants to register', () => {
    RegisterPage.visit();
    
});

When('filling in the required information', () => {
    RegisterPage.fillData('JUANA','juana123','juana123');
});

And('all fields contain valid information', () => {
    RegisterPage.fillData('JUANA','juana123','juana123');
});

Then('load the "confirmation" page', () => {
    RegisterPage.fillData('KJUANA','d','d');
});
~~~

when I run this on cypress, the problem I have is that it always run the code that is in the first file for the 2 steps that the scenarios have in common.

Mabye you are thinking "ok so just put this 2 steps as a background and that's it", but
the issue is that the step of 'filling in the required information' has different values as you see, because one scenario is going to fill with valid information and the other is going to fill with invalid information.
### Desired behavior
I want this to run just in the scenario that corresponds, so in this case I don't want cypress to run always the function of 'filling in the required information' of the first file, when I'm in other scenario.

### Versions

* **Cypress version**: 4.12.1
* **Preprocessor version**: 2.5.4
* **Node version**:

**Comment by bezba** [comment]
in this case, change the gherkin and make them unique. this way it has to run

**Comment by lgandecki** [comment]
We already already unorthodox with the way we allow different features to implement the same step definitions differently. I've done it because I worked on codebases with hundreds of feature files and worrying about collisions between them was very problematic (initially you can write beautifully and easily structured feature files, but later on you are tiptoing around already existing step definitions). For scenarios I don't see that problem as you won't have hundreds of scenarios in a single feature file and the collisions shouldn't be a problem.

For a scenario I think it's fair to say that if two scenarios have the same step, the reader assumes it means the same thing should happen for both of them.
In other words, just change the wording so it's obvious that the first one fills valid and second fills invalid :)
